app/api/server/v1/users.js

```diff
@@ -28,6 +28,229 @@ import { resetTOTP } from '../../../2fa/server/functions/resetTOTP';
+			requirePasswordChange: Match.Maybe(Boolean),
+			setRandomPassword: Match.Maybe(Boolean),
+			sendWelcomeEmail: Match.Maybe(Boolean),
+			verified: Match.Maybe(Boolean),
+			customFields: Match.Maybe(Object),
+		});
+
+		// New change made by pull request #5152
+		if (typeof this.bodyParams.joinDefaultChannels === 'undefined') {
+			this.bodyParams.joinDefaultChannels = true;
+		}
+
+		if (this.bodyParams.customFields) {
+			validateCustomFields(this.bodyParams.customFields);
+		}
+
+		const newUserId = saveUser(this.userId, this.bodyParams);
+
+		if (this.bodyParams.customFields) {
+			saveCustomFieldsWithoutValidation(newUserId, this.bodyParams.customFields);
+		}
+
+
+		if (typeof this.bodyParams.active !== 'undefined') {
+			Meteor.runAsUser(this.userId, () => {
+				Meteor.call('setUserActiveStatus', newUserId, this.bodyParams.active);
+			});
+		}
+
+		const { fields } = this.parseJsonQuery();
+
+		if (this.bodyParams.status) {
+			const validStatus = ['online', 'away', 'offline', 'busy'];
+			if (validStatus.includes(this.bodyParams.status)) {
+				const { status } = this.bodyParams;
+
+				if (status === 'offline' && !settings.get('Accounts_AllowInvisibleStatusOption')) {
+					throw new Meteor.Error('error-status-not-allowed', 'Invisible status is disabled', {
+						method: 'users.setStatus',
+					});
+				}
+
+				Meteor.users.update(newUserId, {
+					$set: {
+						status,
+						statusDefault: status,
+					},
+				});
+
+				setUserStatus(Users.findOneById(newUserId), status);
+			} else {
+				throw new Meteor.Error('error-invalid-status', 'Valid status types include online, away, offline, and busy.', {
+					method: 'users.setStatus',
+				});
+			}
+		}
+
+		return API.v1.success({ user: Users.findOneById(newUserId, { fields }) });
+	},
+});
+
+API.v1.addRoute('users/:userId', { authRequired: true }, {
+	get() {
+		const { userId } = this.urlParams;
+		const { fields } = this.parseJsonQuery();
+
+		const user = getFullUserDataByIdOrUsername({ userId: this.userId, filterId: userId, filterUsername: undefined });
+
+		if (!user) {
+			return API.v1.failure('User not found.');
+		}
+		const myself = user._id === this.userId;
+		if (fields.userRooms === 1 && (myself || hasPermission(this.userId, 'view-other-user-channels'))) {
+			user.rooms = Subscriptions.findByUserId(user._id, {
+				fields: {
+					rid: 1,
+					name: 1,
+					t: 1,
+					roles: 1,
+					unread: 1,
+				},
+				sort: {
+					t: 1,
+					name: 1,
+				},
+			}).fetch();
+		}
+
+		return API.v1.success({
+			user,
+		});
+	},
+	delete() {
+		if (!hasPermission(this.userId, 'delete-user')) {
+			return API.v1.unauthorized();
+		}
+
+		const { userId } = this.urlParams;
+		const { confirmRelinquish = false } = this.requestParams();
+
+		Meteor.runAsUser(this.userId, () => {
+			Meteor.call('deleteUser', userId, confirmRelinquish);
+		});
+
+		return API.v1.success();
+	},
+	put() {
+		check(this.bodyParams, {
+			data: Match.ObjectIncluding({
```

Comment:
@ear-dev @AlexanderKanakis We have added `status` in `users.create` and `users.register`. Should we also have it for updating users(PUT & `users.update`)?

Reply:
Added `status` in PUT. We do not intent to use `users.update`.